src/dxvk/rtx_render/rtx_camera.cpp

```diff
@@ -736,17 +737,79 @@ namespace dxvk
     jitter[1] = m_jitter[1];
   }
 
+  // Static variable to track the actual jitter sequence length being used
+  static uint32_t s_currentJitterSequenceLength = 64;
+  
+  // Static variable to track the current upscaling ratio for dynamic jitter calculation
+  static float s_currentUpscalingRatio = 1.0f;
+
   Vector2 RtCamera::calcPixelJitter(uint32_t jitterFrameIdx) {
-    // Only apply jittering when DLSS/TAA is enabled, or if forced by settings
+    // Only apply jittering when DLSS/XeSS/TAA is enabled, or if forced by settings
     if (!RtxOptions::isDLSSOrRayReconstructionEnabled() &&
+        !RtxOptions::isXeSSEnabled() &&
         !RtxOptions::isTAAEnabled() &&
         !RtxOptions::forceCameraJitter()) {
+      s_currentJitterSequenceLength = 0; // No jitter being used
       return Vector2{ 0, 0 };
     }
 
 #define USE_DLSS_DEMO_JITTER_PATTERN 1
 #if USE_DLSS_DEMO_JITTER_PATTERN
+    uint32_t jitterSequenceLength = RtxOptions::cameraJitterSequenceLength();
+
+    if (RtxOptions::isXeSSEnabled() && DxvkXeSS::xess.useRecommendedJitterSequenceLength()) {
+      float scaleFactor = s_currentUpscalingRatio;
+      uint32_t xessLength = static_cast<uint32_t>(std::ceil(scaleFactor * scaleFactor * 8.0f));
+      
+      // Apply minimum jitter sequence length
+      uint32_t minLength = DxvkXeSS::xess.minJitterSequenceLength();
+      xessLength = std::max(xessLength, minLength);
+      
+      // Clamp to reasonable range
+      xessLength = std::clamp(xessLength, minLength, 1024u);
```

Comment:
Why is this necessary? This is not clamped in other places where the jitter sequence length is used, so I'd keep it consistent.